### PR TITLE
feat: support created and modified methods in GCSFileSystem

### PR DESCRIPTION
**Pull request by Mause**

### CODE REVIEW

File: gcsfs/core.py [hunk 1]
```diff
@@ -741,6 +742,12 @@ async def _rmdir(self, bucket):
 
     rmdir = sync_wrapper(_rmdir)
 
+    def modified(self, path):
+        return datetime.fromisoformat(self.info(path)["updated"])
```
Comment (martindurant) [inline gcsfs/core.py]:
```suggestion
        return datetime.fromisoformat(self.info(path)["updated"].replace(".", ":"))
```

File: gcsfs/core.py [hunk 2]
```diff
@@ -741,6 +742,12 @@ async def _rmdir(self, bucket):
...
+
+    def created(self, path):
+        return datetime.fromisoformat(self.info(path)["timeCreated"])
```
Comment (martindurant) [inline gcsfs/core.py]:
```suggestion
        return datetime.fromisoformat(self.info(path)["timeCreated"].replace(".", ":"))
```

**Comment by Mause** [comment]
Oops, looks like pre 3.11 fromisoformat doesn't support timezones, that's annoying

**Comment by Mause** [comment]
Thankyou!

**Pull request merged**